metadata-load/src/test/java/uk/ac/ebi/ampt2d/metadata/importer/api/SraObjectsImporterThroughAPITest.java

```diff
@@ -16,55 +16,49 @@
  *
  */
 
-package uk.ac.ebi.ampt2d.metadata.importer.objectImporters;
+package uk.ac.ebi.ampt2d.metadata.importer.api;
 
+import org.junit.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.context.junit4.SpringRunner;
 import uk.ac.ebi.ampt2d.metadata.importer.MetadataImporterMainApplication;
-import uk.ac.ebi.ampt2d.metadata.importer.ObjectsImporter;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.Analysis;
 import uk.ac.ebi.ampt2d.metadata.persistence.entities.Study;
 
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
 @RunWith(SpringRunner.class)
 @TestPropertySource(value = "classpath:application.properties", properties = {"import.source=API"})
 @ContextConfiguration(classes = {MetadataImporterMainApplication.class})
-public class SraObjectsImporterAPITest {
+public class SraObjectsImporterThroughAPITest {
 
     @Autowired
-    private ObjectsImporter sraObjectImporter;
+    private SraObjectsImporterThroughAPI sraObjectImporter;
 
     @Test
     public void importStudy() throws Exception {
-        Set<String> studyAccessions = new HashSet<>();
-        studyAccessions.add("ERP000054");
-        studyAccessions.add("SRP000717");
-        List<Study> studies = sraObjectImporter.importStudy(studyAccessions);
-        assertNotNull(studies);
-        assertEquals(2, studies.size());
+        Study study = sraObjectImporter.importStudy("ERP000054");
+        assertNotNull(study);
 
-        studyAccessions.clear();
-        studyAccessions.add("SRP000392");
-        studies = sraObjectImporter.importStudy(studyAccessions);
-        assertNotNull(studies);
-        assertEquals(1, studies.size());
-        assertEquals(1, studies.get(0).getPublications().size());
+        study = sraObjectImporter.importStudy("SRP000392");
+        assertNotNull(study);
+        assertEquals(1, study.getPublications().size());
 
-        studyAccessions.clear();
-        studyAccessions.add("SRP000118");
-        studies = sraObjectImporter.importStudy(studyAccessions);
-        assertNotNull(studies);
-        assertEquals(1, studies.size());
-        assertEquals(1, studies.get(0).getResources().size());
+        study = sraObjectImporter.importStudy("SRP000118");
+        assertNotNull(study);
+        assertEquals(1, study.getResources().size());
     }
```

Comment:
Check at the end that 3 studies have been stored.

Reply:
the studies won't be persisted yet at that moment.

Reply:
reading this comment I realized that the study/analysis should be persisted in importer rather than in MainApplication class. Implementing the same